### PR TITLE
[Communication] - SMS - Adding Repeteability Request ID

### PR DESCRIPTION
**Pull request by paolamvhz**

### CODE REVIEW

**Comment by minnieliu** [comment]
Please update recording so that it can reflect the new headers we are sending to the service

**Comment by JianpingChen** [comment]
The change looks simple. How did it slip though our integration tests? If this is a big bug, can we add an integration test to ensure this is covered?

**Comment by lsundaralingam** [comment]
Lets run the live tests as well

**Comment by minnieliu** [comment]
I think we also need to update the changelog stating the bug fix?

File: sdk/communication/azure-communication-sms/CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Updated `azure-communication-sms` version
 - Added support for tagging SMS messages.
 - Send method series in SmsClient are idempotent under retry policy.
 - Added `SmsOptions`
+- Added implementation of the repeteabilityRequestId.
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-sms/CHANGELOG.md:17]: No. The reality is this was not fixed in 03-39 release but the latest. Move this to be behind 1.1.0-beta.1

**Review by JianpingChen — APPROVED** [review]

**Comment by sacheu** [comment]
Should we add new test cases for this change?

File: sdk/communication/azure-communication-sms/CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Release History
 
 ## 1.1.0-beta.1 (Unreleased)
-
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-sms/CHANGELOG.md:4]: Please put a subheading

### Bug Fixes

**Review by ankitarorabit — APPROVED** [review]

File: sdk/communication/azure-communication-sms/pom.xml [hunk 3]
```diff
@@ -105,6 +105,12 @@
       <version>1.2.5</version> <!-- {x-version-update;com.azure:azure-identity;dependency} -->
       <scope>test</scope>
     </dependency>
+      <dependency>
+          <groupId>com.nimbusds</groupId>
+          <artifactId>nimbus-jose-jwt</artifactId>
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-sms/pom.xml:110]:
```suggestion
          <artifactId>nimbus-jose-jwt</artifactId> <!-- {x-version-update;com.nimbusds:nimbus-jose-jwt;external_dependency}
```

File: sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java [hunk 4]
```diff
@@ -196,6 +199,28 @@ public void sendTwoMessages(HttpClient httpClient) {
...
+        StepVerifier.create(asyncClient.sendWithResponse(FROM_PHONE_NUMBER, Arrays.asList(TO_PHONE_NUMBER, TO_PHONE_NUMBER), MESSAGE, null, Context.NONE)
+        )
+            .assertNext(requestResponse -> {
+                requestResponse.getRequest().getBody().map(bodyBuffer -> {
+                    String bodyRequest = new String(bodyBuffer.array());
+                    assertTrue(bodyRequest.contains("repeatabilityRequestId"));
+                    return null;
+                });
+            }
+            )
```
Comment (minnieliu) [inline sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java:219]: Can you fix the formatting issues here? and make 218 and 219 the same line?

File: sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java [hunk 5]
```diff
@@ -196,6 +199,28 @@ public void sendTwoMessages(HttpClient httpClient) {
...
+            .verifyComplete();
+
```
Comment (minnieliu) [inline sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java:221]: extra space

File: sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java [hunk 6]
```diff
@@ -196,6 +199,28 @@ public void sendTwoMessages(HttpClient httpClient) {
...
+        // Arrange
+        SmsClientBuilder builder = getSmsClientUsingConnectionString(httpClient);
+        asyncClient = setupAsyncClient(builder, "sendTwoMessagesSync");
+
+        StepVerifier.create(asyncClient.sendWithResponse(FROM_PHONE_NUMBER, Arrays.asList(TO_PHONE_NUMBER, TO_PHONE_NUMBER), MESSAGE, null, Context.NONE)
+        )
+            .assertNext(requestResponse -> {
+                requestResponse.getRequest().getBody().map(bodyBuffer -> {
+                    String bodyRequest = new String(bodyBuffer.array());
+                    assertTrue(bodyRequest.contains("repeatabilityRequestId"));
```
Comment (minnieliu) [inline sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java:215]: We should probably add a check for the RepeatabilityFirstSent as well

File: sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsClientTests.java [hunk 7]
```diff
@@ -165,6 +166,21 @@ public void sendTwoMessages(HttpClient httpClient) {
...
+        PipedOutputStream osPipe = new PipedOutputStream();
+        // Arrange
+        SmsClientBuilder builder = getSmsClientUsingConnectionString(httpClient);
+        client = setupSyncClient(builder, "sendTwoMessagesSync");
+
+        // Action & Assert
+        Response<Iterable<SmsSendResult>> response = client.sendWithResponse(FROM_PHONE_NUMBER, Arrays.asList(TO_PHONE_NUMBER, TO_PHONE_NUMBER), MESSAGE, null, Context.NONE);
+        String bodyRequest = new String(response.getRequest().getBody().blockLast().array());
+        assertTrue(bodyRequest.contains("repeatabilityRequestId"));
+
```
Comment (minnieliu) [inline sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsClientTests.java:181]: Extra lines

Comment (minnieliu) [inline sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsClientTests.java:181]: We should probably add a check for the RepeatabilityFirstSent as well

**Review by minnieliu — CHANGES_REQUESTED** [review]

File: sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsClientTests.java (hunk 7 above)
Comment (beltr0n) [reply sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsClientTests.java:181]: we should probably also validate that repeatabilityRequestId value follows a guid format if were going to make a test for it

File: sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java [hunk 8]
```diff
@@ -196,6 +199,28 @@ public void sendTwoMessages(HttpClient httpClient) {
...
+    @ParameterizedTest
+    @MethodSource("com.azure.core.test.TestBase#getHttpClients")
+    public void repeatability(HttpClient httpClient) {
+        PipedOutputStream osPipe = new PipedOutputStream();
+        // Arrange
+        SmsClientBuilder builder = getSmsClientUsingConnectionString(httpClient);
+        asyncClient = setupAsyncClient(builder, "sendTwoMessagesSync");
+
+        StepVerifier.create(asyncClient.sendWithResponse(FROM_PHONE_NUMBER, Arrays.asList(TO_PHONE_NUMBER, TO_PHONE_NUMBER), MESSAGE, null, Context.NONE)
+        )
```
Comment (minnieliu) [inline sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java:211]: Formatting issue

File: sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java [hunk 9]
```diff
@@ -196,6 +199,28 @@ public void sendTwoMessages(HttpClient httpClient) {
             .verifyComplete();
     }
 
+    @ParameterizedTest
+    @MethodSource("com.azure.core.test.TestBase#getHttpClients")
+    public void repeatability(HttpClient httpClient) {
+        PipedOutputStream osPipe = new PipedOutputStream();
+        // Arrange
+        SmsClientBuilder builder = getSmsClientUsingConnectionString(httpClient);
+        asyncClient = setupAsyncClient(builder, "sendTwoMessagesSync");
```
Comment (beltr0n) [inline sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java:208]: is "sendTwoMessagesSync" the correct param for this method?

File: sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsClientTests.java (hunk 7 above)
Comment (paolamvhz) [reply sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsClientTests.java]: Just validated the words repeatabilityRequestId  and repeatabilityFirstSent

**Review by minnieliu — APPROVED** [review]
Looks good!!

**Pull request merged**